File: Room.cs
using System; 
namespace DungeonExplorer
{
    public class Room
    {
        private string description;  // Private field for room description
        private string item;         // Private field for the item in the room (could be null if no item is present)

        // Constructor to initialize Room with a description and an item (optional)
        public Room(string description, string item = null)
        {
            this.description = description;
            this.item = item;  // If no item is passed, it will default to null
        }

        // Method to get the room's description
        public string GetDescription()
        {
            return description;
        }

        // Method to check if the room contains an item
        public bool HasItem(string itemName)
        {
            return item != null && item.Equals(itemName, StringComparison.OrdinalIgnoreCase);
        }

        // Method to remove the item from the room once it's picked up
        public void RemoveItem()
        {
            item = null;  // Remove the item once it's picked up
        }

        // Method to get the item in the room
        public string GetItem()
        {
            return item;
        }
    }
}

File: room.py
class Room:
    def __init__(self, description, item=None):
        self.__description = description
        self.__item = item  # Item available in the room

    def get_description(self):
        """Returns the room's description."""
        return self.__description

    def get_item(self):
        """Returns the item in the room (if any)."""
        return self.__item

    def remove_item(self):
        """Removes the item from the room."""
        self.__item = None